runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -9061,7 +9061,7 @@ generateCheckForValueMonitorEnterOrExit(TR::Node *node, TR::LabelSymbol* mergeLa
       helperCallOOLSection->swapInstructionListsWithCompilation();
       }
 
-   cg->stopUsingRegister(tempReg);
+   srm->reclaimScratchRegister(tempReg);
    return objectClassReg;
```

Comment:
not sure if it is a good idea to return a scratch register!

Reply:
Unless and untill that register is reclaimed, I do not see why this is a problem. It is a manager for scratch register, but underlying mechanism for allocating registers are same. So Should not be the issue.